src/ui/app/ApiProvider.ts

```diff
@@ -148,12 +149,13 @@ export const generateActiveNetworkList = (): NetworkTypes[] => {
 export default class ApiProvider {
     public fallbackNumber: number | undefined = undefined;
 
-    private _apiFullNodeProvider?: JsonRpcProvider;
+    private _apiFullNodeClient?: SuiClient;
     private _signer: RawSigner | null = null;
     private _apiEnv: API_ENV = DEFAULT_API_ENV;
     private _customRPC: string | null = null;
+    private _connection: Connection | null = null;
 
-    public setNewJsonRpcProvider(
+    public setNewSuiClient(
```

Comment:
This can eventually be refactored to remove all things referencing `Connection` etc., but it's complicated so I've just left it untouched for now since this is already a large PR

Reply:
Update: I did this in `ag/apiprovider`, which can be merged onto this or separately after merging this to master